CRM/Event/Form/Registration.php

```diff
@@ -403,14 +403,23 @@ public function preProcess() {
       $this->_isBillingAddressRequiredForPayLater = $this->_values['event']['is_billing_required'] ?? NULL;
       $this->assign('isBillingAddressRequiredForPayLater', $this->_isBillingAddressRequiredForPayLater);
     }
+
+    // set the noindex metatag for non-public events
+    if (!$values['event']['is_public']) {
+      CRM_Utils_System::addHTMLHead('<META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">');
+    }
+
   }
 
   /**
    * Assign the minimal set of variables to the template.
    */
   public function assignToTemplate() {
     //process only primary participant params
-    $this->_params = $this->get('params');
```

Comment:
I think it was a mistake that this line was moved.

Reply:
my mistake, thanks for spotting!